pkg/kopia/command/storage/secret_utils.go

```diff
@@ -57,6 +74,94 @@ func locationType(m map[string][]byte) LocType {
 	return LocType(m[typeKey])
 }
 
+// GenerateEnvSpecFromCredentialSecret parses the secret and returns
+// list of EnvVar based on secret type
+func GenerateEnvSpecFromCredentialSecret(s *v1.Secret) ([]v1.EnvVar, error) {
+	if s == nil {
+		return nil, errors.New("Secret cannot be nil")
+	}
+	secType := string(s.Type)
+	switch secType {
+	case secrets.AWSSecretType:
+		return getEnvSpecForAWSCredentialSecret(s)
+	case secrets.AzureSecretType:
+		return getEnvSpecForAzureCredentialSecret(s)
+	}
```

Comment:
What about location secrets of type GCS and NFS stores?

Reply:
For GCS, we would use the creds file and we don't have required ENV available.
For NFS, we don't need the credentials.